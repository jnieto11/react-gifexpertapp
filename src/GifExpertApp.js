import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {


    const [categories, setcategories] = useState(['Dragon Ball']);


    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setcategories ={ setcategories}></AddCategory>
            <hr />

            

            <ol>
                {
                    categories.map(category => {
                     return  <GifGrid key={category} category={category} />
                    })
                }
            </ol>
        </>
    )
}
