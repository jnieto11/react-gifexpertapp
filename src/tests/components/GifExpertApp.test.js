import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import '@testing-library/jest-dom';
import { shallow, configure } from 'enzyme';
import { GifExpertApp } from "../../GifExpertApp";

configure({ adapter: new Adapter() });

describe('Prueba con GifExpertApp', () => {

    test('Debe mostrarse correctamente ', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar una lista de categories', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
