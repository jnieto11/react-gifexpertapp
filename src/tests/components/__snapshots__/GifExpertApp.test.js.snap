// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con GifExpertApp Debe mostrar una lista de categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Prueba con GifExpertApp Debe mostrarse correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
